pallets/aleph/src/lib.rs

```diff
@@ -121,6 +142,57 @@ pub mod pallet {
         pub(crate) fn set_next_emergency_finalizer(emergency_finalizer: T::AuthorityId) {
             <NextEmergencyFinalizer<T>>::put(emergency_finalizer);
         }
+
+        pub(crate) fn current_session() -> u32 {
+            T::SessionInfoProvider::current_session()
+        }
+
+        // If a scheduled future version change is rescheduled to a different session,
+        // it should be possible to reschedule it with the same version as initially.
+        // If a scheduled version change has moved into the past, `SessionManager` records it
+        // as the current version.
+        pub(crate) fn schedule_next_aleph_bft_version_change(
+            version_change: VersionChange,
+        ) -> Result<(), &'static str> {
+            let current_session = Self::current_session();
+
+            let session_to_schedule = version_change.session;
+
+            if session_to_schedule < current_session {
+                return Err("Cannot schedule AlephBFT version changes for sessions in the past!");
+            } else if session_to_schedule < current_session + 2 {
+                return Err(
+                    "Tried to schedule AlephBFT version change less than 2 sessions in advance!",
+                );
+            }
+
+            // Update the scheduled version change with the supplied version change.
+            <AlephBFTScheduledVersionChange<T>>::put(version_change);
+
+            Ok(())
+        }
+
+        pub(crate) fn remove_next_aleph_bft_version_change() -> Result<VersionChange, &'static str>
+        {
+            let version_change_to_remove = Self::aleph_bft_version_change()
+                .ok_or("No AlephBFT version change to unschedule!")?;
+            <AlephBFTScheduledVersionChange<T>>::kill();
```

Comment:
You cannot do that when the session is too close, requires a check as in the other method.

But also, since you removed the identity check above maybe just remove this method to lower the bug surface? Sorry for suggesting opposite things in two different comments, I actually don't have an opinion on which you should do but please pick one. ^^"

Reply:
I was on the fence for this one, since the remover was more explicit but it resulted in a bit of bloat. The more implicit approach, however, results in a simpler implementation and I tend to agree with the remark about the bug surface. So, the remover is gone, and we can cancel a scheduled version change via a rescheduling with the same version as the one currently recorded.